src/coreclr/src/vm/class.cpp

```diff
@@ -1184,107 +1184,112 @@ void ClassLoader::ValidateMethodsWithCovariantReturnTypes(MethodTable* pMT)
         return;
 
     // Step 1: Validate compatibility of return types on overriding methods
-
-    for (WORD i = 0; i < pParentMT->GetNumVirtuals(); i++)
+    if (pMT->GetClass()->HasCovariantOverride())
```

Comment:
We may skip it when `pModule->GetReadyToRunInfo()->SkipTypeValidation()` is set. It is what we do for similar validation checks in other places.

Reply:
Hmm, yes, I could probably add that check here, so that even if we have a covariant override, that we don't need to run this code.